In [1]:
import pandas as pd
import numpy as np

# Jawaban Nomor 1

In [2]:
data = {
    'Tipe Ujian':['UTS1', 'UAS1'] * 3,
    'Mata Pelajaran':['Statistika', 'Fisika', 'Fisika', 'Statistika', 'Fisika', 'Statistika'],
    'Jumlah Peserta':[14, 17, 12, 16, 14, 13]
}

data = pd.DataFrame(data).set_index('Tipe Ujian')
data

Mata Pelajaran  Jumlah Peserta
Tipe Ujian                               
UTS1           Statistika              14
UAS1               Fisika              17
UTS1               Fisika              12
UAS1           Statistika              16
UTS1               Fisika              14
UAS1           Statistika              13

In [8]:
data.pivot_table(index='Tipe Ujian', columns='Mata Pelajaran', values='Jumlah Peserta', aggfunc=sum)

Mata Pelajaran  Fisika  Statistika
Tipe Ujian                        
UAS1                17          29
UTS1                26          14

# Jawaban Nomor 2

In [12]:
df = {
    'Hari':['Sabtu', 'Minggu'],
    'Kebun Binatang':[271, 399],
    'Dufan':[501, 700],
    'Disney':[1000, 1001],
    'Bali':[900, 803]
}

df = pd.DataFrame(df).set_index('Hari')
df

Kebun Binatang  Dufan  Disney  Bali
Hari                                       
Sabtu              271    501    1000   900
Minggu             399    700    1001   803

In [14]:
df.reset_index().melt(id_vars='Hari', var_name='Tempat Hiburan',
                      value_name='Pengunjung', value_vars=['Kebun Binatang', 'Dufan', 'Disney', 'Bali'])

Hari  Tempat Hiburan  Pengunjung
0   Sabtu  Kebun Binatang         271
1  Minggu  Kebun Binatang         399
2   Sabtu           Dufan         501
3  Minggu           Dufan         700
4   Sabtu          Disney        1000
5  Minggu          Disney        1001
6   Sabtu            Bali         900
7  Minggu            Bali         803

# Jawaban Nomor 3

In [16]:
df_diamond = pd.read_csv('/content/sample_data/diamonds-1.csv')

df_diamond

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

In [20]:
def min_max(x):
  return x.max() - x.min()

df_diamond.groupby('cut')[['price', 'carat']].agg({'price':min_max, 'carat':'std'})

price     carat
cut                       
Fair       18237  0.516404
Good       18461  0.454054
Ideal      18480  0.432876
Premium    18497  0.515262
Very Good  18482  0.459435

jawaban :

***- Selisih harga tertinggi dan terendah dari diamond berkualitas Premium adalah 18497***